caom2pipe/caom2pipe/execute_composable.py

```diff
@@ -0,0 +1,1163 @@
+execution for an Observation.
+
+The correlations that currently exist:
+- use_local_data: True => classes have "Local" in their name
+- uses the CadcDataClient, the Caom2RepoClient => classes have "Client"
+    in their name
+- requires metadata access only => classes have "Meta" in their name
+- requires data access => classes have "Data" in their name
+
+"""
+
+import logging
+import os
+import re
+import site
+import traceback
+
+from argparse import ArgumentParser
+from datetime import datetime
+
+from cadcutils import net, exceptions
+from cadcdata import CadcDataClient
+from caom2repo import CAOM2RepoClient
+from caom2pipe import manage_composable as mc
+
+
+__all__ = ['OrganizeExecutes', 'StorageName', 'CaomName']
+
+
+class StorageName(object):
+    """Naming rules:
+    - support mixed-case file name storage
+    - support gzipped and not zipped file names
+
+    This class assumes the obs_id is part of the file name. This assumption
+    may be broken in the future, in which case lots of CaomExecute
+    implementations will need to be re-addressed somehow.
+
+    This class assumes the file name in storage, and the file name on disk
+    are not necessarily the same thing.
+    """
+
+    def __init__(self, obs_id, collection, collection_pattern,
+                 fname_on_disk=None):
+        self.obs_id = obs_id
+        self.collection = collection
+        self.collection_pattern = collection_pattern
+        self.fname_on_disk = fname_on_disk
+
+    def get_file_uri(self):
+        return 'ad:{}/{}.gz'.format(self.collection, self.get_file_name())
+
+    def get_file_name(self):
+        return '{}.fits'.format(self.obs_id)
+
+    def get_compressed_file_name(self):
+        return '{}.fits.gz'.format(self.obs_id)
+
+    def get_model_file_name(self):
+        return '{}.fits.xml'.format(self.obs_id)
+
+    def get_prev(self):
+        return '{}_prev.jpg'.format(self.obs_id)
+
+    def get_thumb(self):
+        return '{}_prev_256.jpg'.format(self.obs_id)
+
+    def get_prev_uri(self):
+        return self._get_uri(self.get_prev())
+
+    def get_thumb_uri(self):
+        return self._get_uri(self.get_thumb())
+
+    def get_obs_id(self):
+        return self.obs_id
+
+    def get_log_file(self):
+        return '{}.log'.format(self.obs_id)
+
+    def get_product_id(self):
+        return self.obs_id
+
+    def get_fname_on_disk(self):
+        return self.fname_on_disk
+
+    def is_valid(self):
+        pattern = re.compile(self.collection_pattern)
+        return pattern.match(self.obs_id)
+
+    def _get_uri(self, fname):
+        return 'ad:{}/{}'.format(self.collection, fname)
+
+    @staticmethod
+    def remove_extensions(name):
+        return name.replace('.fits', '').replace('.gz', '').replace('.header',
+                                                                    '')
+
+
+class CaomName(object):
+    """The naming rules for making and decomposing CAOM URIs, all isolated in
```

Comment:
StorageName is collection <==> ad.
CaomName is caom2 <==> collection components, where caom2 is mostly Observation URIs, Plane URIs, and artifact URIs.

They shouldn't contain the same thing ever. It just looks that way, because file names are involved. I added to the CaomName class description.